src/shaders/globe_raster.fragment.glsl

```diff
@@ -1,11 +1,46 @@
 uniform sampler2D u_image0;
 varying vec2 v_pos0;
 
+#ifndef FOG
+uniform vec3 u_frustum_tl;
+uniform vec3 u_frustum_tr;
+uniform vec3 u_frustum_br;
+uniform vec3 u_frustum_bl;
+uniform highp vec3 u_globe_pos;
+uniform highp float u_globe_radius;
+uniform vec2 u_viewport;
+#endif
+
+uniform bool u_is_antialias;
+
 void main() {
+#ifdef GLOBE_POLES
     vec4 color = texture2D(u_image0, v_pos0);
+#else
+    if (u_is_antialias) {
+        vec2 uv = gl_FragCoord.xy / u_viewport;
+
+        vec3 ray_dir = mix(
+            mix(u_frustum_tl, u_frustum_tr, uv.x),
+            mix(u_frustum_bl, u_frustum_br, uv.x),
+            1.0 - uv.y);
+        
+        vec3 dir = normalize(ray_dir);
 
+        vec3 closest_point = dot(u_globe_pos, dir) * dir;
+        float norm_dist_from_center = 1.0 - length(closest_point - u_globe_pos) / u_globe_radius;
+
+        const float antialias_distance_px = 4.0;
+        float antialias = smoothstep(0.0, antialias_distance_px / max(u_viewport.x, u_viewport.y), norm_dist_from_center);
+
+        vec4 raster = texture2D(u_image0, v_pos0);
+        vec4 color = vec4(raster.rgb * antialias, raster.a * antialias);
+    } else {
+        vec4 color = texture2D(u_image0, v_pos0);
```

Comment:
when do we need this branch?

Reply:
In the case that custom antialias isn't needed for globe edges because antialias is set to true in the map constructor

Reply:
I meant, is the additional complexity required? What happens if both antialiasing are enabled?

Reply:
Also, else branch of `if (u_is_antialias) {` seems to be the same code as inside

```
#ifdef GLOBE_POLES
    vec4 color = texture2D(u_image0, v_pos0);
#else
```

did you reconsider using a define for both instead of uniform branch?

Reply:
Thanks @astojilj for the feedback, I swapped with using a define.